### ISSUE TITLE
Fight between bSpace 2.3 and multiverse core 2.4

### ISSUE THREAD

**Issue by Gnarler** [kitskub/bSpace#111]
I am able to get bSpace to list when I request the list of generators, however when I generate a space world using schematics and planets it lags for a very long time, and then presents me with a "internal error" generic fail.   Multiverse won't recognize it  with /mv tp space, but if I use /space enter I can go there.  

After some time the server will randomly crash, and when restarted not even bSpace will recognize the world.  I see it in the folder, the files are there, but what looks like a header from both bSpace and multiverse has been removed so people cant go there.

Observe the pastebin.  http://pastebin.com/MARWMtbn

I noticed an out of memory crash when I used BukkitNyan, a movement plugin that never caused crashes before.  The server was recently upgraded to have 1 gig of memory and is running 1.2.5 git-Bukkit-1.2.5-R1.0-b2149jnks (MC: 1.2.5) 
 bSpace 2.3 and multiverse core 2.4

I heard an older version of the 2 plugins worked better together... could you please tell me which versions work well together?

Also where does bSpace leave the list of spaceworlds? I RTFM'd but still cant find it.  

So in conclusion when the world is first made, multiverse won't recognize it, and later the world itself looses its "space header" and ability to warp to.  Even warps made before the crash are impossible to go to now.

**Comment by kitskub** [comment]
My best guess is that this is multiverse. I'm seeing a nullpointerexception in there with the last line of multiverse; this isn't confirmation that it's multiverse, but it most likely is. Try downgrading multiverse, it might help. And, bSpace does not store spaceworlds. I will continue to look into this.

**Comment by kitskub** [comment]
Oh, and if you did /reload, don't do that, it creates memory leaks.

**Comment by Gnarler** [comment]
Heres another one  http://pastebin.com/N6gjK4wN

out of memory error for schematics, and I have some large schematics but I can paste them all normally with only slight lag, and they are under 1M blocks.

Multiverse 2.3 is for 1.1 and does not generate a plugin folder or initialize.

**Comment by Gnarler** [comment]
I got it working with multiworld.  Please change the informational post to reccommend multiworld.  
the command is /mw create (name) plugin:bSpace

**Comment by Gnarler** [comment]
I take that back.  I wanted to make a change in the settings and regen a new space, and after a restart I tried to gen another one and all I get is errors.

So barring multiworld (refuses to work, error message blames bspace) and multiverse (plugin creators have over 100 unsolved tickets and world is corrupted after 10 or so minutes) is there ANY way to get a generated world using this plugin?  I am dying to get it working and stable.

Also, apparently no matter how I change the settings, even if I want an empty world, the generated world has asteroids and the time isn't stationary.

Thank you for your time.  Could you recommend to me another multiple world plugin?

**Comment by kitskub** [comment]
Ok, so, for the schematic stuff, I am looking at WorldEdit code, because the code we have is slightly different from that, yet that is where we originally got this code. Should be able to fix that.

Can you post the errors you get when you use multiworld? It might help to find a common link.

**Comment by Gnarler** [comment]
This is a good example.

http://pastebin.com/F34n7zmC

The paste starts when I type /mw create spaceworld plugin:bSpace.   bSpace is a fresh install with schematics in the schematic folder.  it Should create an empty world because I have not changed any settings.

First the in game error says there was an error with the generator,  and then when I type the same command again, it goes ahead and generates something partially but then dies and I am unable to "enable" the spaceworld.

**Comment by kitskub** [comment]
Make sure you are using the latest: https://github.com/iffa/bSpace/blob/master/target/bSpace-v2.3.jar
And, by default, the generator is not empty.

**Comment by Gnarler** [comment]
O-k I downloaded the new version 2.3 and the filesize was a few k different from the version 2.3 I had so I replaced it and will test it out.

A previous crash resulted in this 
http://pastebin.com/23WsUB5n
(still 2.3 but not the one you offered me)

Also, the reason I said empty generator was because in the ids file when the plugin first loads there is only one id and that is empty with everything set to false. Apparently no matter how I modify this file it dosn't matter, the schematics and planets occur at the same probibility.

**Comment by Gnarler** [comment]
Cool. New version is alot smoother, no errors when generating. 
However... schematics wont paste in the space worlds.  Still seems to ignore the ids config file.  I'm using /mw create spaceworld plugin:bSpace. No errors to pastebin to you, except at the finish of the generation it said "system time can't keep up.

I wonder why the schematics are being ignored now?

**Comment by kitskub** [comment]
I will check schematics being ignored. Again though, by default bSpace uses a generator that is not empty. If you want an empty generat>r, you must make an id the use bSpace:<name>

**Comment by Gnarler** [comment]
Ok well I have been able to generate many space worlds but it still seems to ignore the config on at least density.  I can change the planet size and materials but the density and schematics are wonky.  Also, during a reset, the space world lost its header and I was unable to access it.  Its hard to reproduce because its not consistant, but if it happens again I'll paste here.

**Comment by Gnarler** [comment]
Ok the new version you sent me works like a charm, both multiverse and multiworld can use it.
however... heres the issue with the schematics.
http://pastebin.com/pqQvqMC3
My schematics are mce not gzip.  Whats the difference from mce and mcedit?  

Also the spawn chunk enderportal blocks keep generating and soon make a pillar of endstone.
Thanks.

**Comment by Gnarler** [comment]
Please notice the Gzip problem. Theres no way for schematics to load now using bSpace.

**Comment by kitskub** [comment]
Sorry. The past couple days have been super hectic. I will try to figure out the best way to load these. Not sure what it is.

**Comment by Gnarler** [comment]
It seemed that schematics could load in earlier versions but didn't work with the multiverse plugins... I wonder if the two errors are connected?

**Comment by mattwebbio** [comment]
I'm having the same problem with schematics being loaded - in the console I get:
"[WARNING] [bSpace] There was a problem while loading schematic file '(Filename).schematic'! Are you sure it's a schematic: Not in GZIP format"

**Comment by kitskub** [comment]
This issue has not been solved, nor do I know what it could be. However, I do not have time to work on bSpace right now. I hope I do have time to get back to it, though.
